File: src/gestorAplicación/clasesPrincipales/Pago.java
package gestorAplicación.clasesPrincipales;
import java.io.Serializable;
import java.util.ArrayList;

public class Pago implements Serializable {
    private Factura factura;
    private String medioPago;
    public static ArrayList<Factura> facturasPagas = new ArrayList<Factura>();
    public static ArrayList<Factura> facturasPendientes = new ArrayList<Factura>();
    public static ArrayList<Pago> registroPagos = new ArrayList<Pago>();

    public Pago(Factura factura, String medioPago) {
    	this.factura = factura;
    	this.medioPago = medioPago;
        registroPagos.add(this);  
    }
    public static void addFacturasPagas(Factura factura) {
    	facturasPagas.add(factura);
    }
    public static ArrayList<Factura> getFacturasPagas() {
    	return facturasPagas;
    }
    public static void removePaga(Factura factura) {
    	facturasPagas.remove(factura);
    }
   
    public static void addFacturasPendientes(Factura factura) {
    	facturasPendientes.add(factura);
    }
    public static ArrayList<Factura> getFacturasPendientes() {
    	return facturasPendientes;
    }
    public static void removePendiente(Factura factura) {
    	facturasPendientes.remove(factura);
    }
    public Factura getFactura() {
    	return factura;
    }
    public String getMedioPago() {
    	return medioPago;
    }
    public static ArrayList<Pago> getRegistroPagos(){
    	return registroPagos;
    }
    public static void removePagos(Pago pago) {
    	registroPagos.remove(pago);
    }
}

File: Python/src/gestorAplicación/clasesPrincipales/Pago.py
class Pago:
    facturasPagas = []
    facturasPendientes = []
    registroPagos = []

    def __init__(self, factura, medioPago):
        self._factura = factura
        self._medioPago = medioPago
        self.registroPagos.append(factura)


    #Métodos get
    @staticmethod
    def getFactuasPagas():
        return Pago.facturasPagas
    
    @staticmethod
    def getFactuasPendientes():
        return Pago.facturasPendientes
    
    @staticmethod
    def getRegistroPagos():
        return Pago.registroPagos
    
    def getFactura(self):
        return self._factura
    
    def getMedioPago(self):
        return self._medioPago
    
    #Otros métodos
    @staticmethod
    def addFacturasPagas(factura):
        Pago.facturasPagas.append(factura)

    @staticmethod
    def removePaga(factura):
        Pago.facturasPagas.remove(factura)
    
    @staticmethod
    def addFacturasPendientes(factura):
        Pago.facturasPendientes.append(factura)
    
    @staticmethod
    def removePendiente(factura):
        Pago.facturasPendientes.remove(factura)

    @staticmethod
    def removePagos(pago):
        Pago.registroPagos.remove(pago) 